### PR TITLE
`parse_reg()`: Avoid unnecessary reg clash messages

### LINKED ISSUE #4358: Centralised reg parse should handle "./" better
Follow-up to #4285 

The centralised reg parse should recognise that `./<reg>` means only look in cwd for `<reg>`, not in `~/cylc-run`. But with #4285 you get an unnecessary warning:
```
$ cd ~/cylc-src
$ cylc validate ./armageddon 
WARNING - The specified reg could refer to ./armageddon/flow.cylc or ~/cylc-run/armageddon/run1/flow.cylc.
This command will use ./armageddon/flow.cylc.
```

### PR DESCRIPTION
**Pull request by MetRonnie**
These changes close #4358 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.py` and `conda-environment.yml`.
- [x] Appropriate tests are included (unit)
- [x] No changelog entry needed (fixing an annoyance introduced in #4285, we didn't release a version since that)
- [x] No documentation update required.

### CODE REVIEW

**Review by hjoliver — APPROVED** [review]
LGTM :+1:

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -1169,10 +1169,11 @@ def parse_reg(reg: str, src: bool = False, warn_depr=True) -> Tuple[str, Path]:
     """
     if not src:
         validate_workflow_name(reg)
+    cur_dir = reg.startswith(f'{os.curdir}{os.sep}')  # if starts with './'
```
Comment (wxtim) [inline cylc/flow/workflow_files.py:1172]: What is this short for?
I guess it's current directory, but if it is, that doesn't really match a data type of boolean IMO.
It's a bit of a mouthful, but I thing `reg_is_curdir` might be more helpful?

**Review by wxtim — APPROVED** [review]
- [x] Read and happy
- [x] Tried it locally

Made a small written communication suggestion. Matter of opinion which shouldn't stand in the way of merger if you wish to leave as is.

**Comment by MetRonnie** [comment]
(Do remember to wait for the tests to pass before merging)

**Comment by wxtim** [comment]
> (Do remember to wait for the tests to pass before merging - in this case failing due to flakiness so ok)

I was confident that they would pass after the change you had made, because the change was not significant, and they passed before.

**Pull request merged**